lib/provider-search.js

```diff
@@ -96,5 +54,53 @@ const isProviderAllowed = (providerEndpoint) => {
   return false;
 };
 
-module.exports = isProviderAllowed;
+const refreshProvider = () => {
+  whitelistEndpoints = [];
+  if (providerJSONFileExists) {
+    console.log(`Reading providers JSON from ${process.env.PROVIDER_JSON_FILE} file.`);
+
+    try {
+      whitelistEndpoints = JSON.parse(fs.readFileSync(process.env.PROVIDER_JSON_FILE, 'utf8'));
+      whitelistEndpoints.sort();
+    } catch (e) {
+      console.error(`Error reading providers JSON: ${e}`);
+    }
+  } else {
+    client = new pg.Client({
+      user: process.env.POSTGRES_USER,
+      password: process.env.POSTGRES_PASSWORD,
+      host: process.env.DBHOST,
+      port: process.env.DBPORT,
+      database: process.env.POSTGRES_DB,
+    });
+
+    client.connect().then(() => {
+      console.log('Connected to PostgreSQL database');
+
+      const query = {
+        name: 'fetch-providers',
+        text: 'SELECT "FHIR_api_base_uri" FROM core_provider where use_proxy = true',
+      };
+
+      client.query(query).then((result) => {
+        result.rows?.forEach(row => {
+          whitelistEndpoints.push(row['FHIR_api_base_uri']);
+        });
+        whitelistEndpoints.sort();
+      }).catch((err) => {
+        console.error(err);
+      });
+
+    }).catch((err) => {
+      console.error('Error connecting to PostgreSQL database', err);
+    });
+  }
```

Comment:
we don't want this proxy to have privileged access to piece.  I'm sorry that was not clear.

The endpoint can merely can take in a POSTed JSON payload.  The payload is JSON array of URLs to add to the whitelist database (in this case, that's our JSON file on the filesystem).

The endpoint will need to be protected via some mechanism, a simple Basic Auth user/password "admin user"

Chad can probably write the code if you ask it.

Let's discuss.